test/unit/org/apache/cassandra/service/paxos/ContentionStrategyTest.java

```diff
@@ -427,4 +428,39 @@ public void boundParseFailureTest()
         assertParseFailure("p95");
         assertParseFailure("p50(rw)+0.66");
     }
+
+    @Test
+    public void testBackoffTime()
+    {
+        ContentionStrategy strategy = parseStrategy("min=0ms,max=100ms,random=uniform").strategy;
+        double total = 0;
+        int count = 100000;
+        for (int i = 0 ; i < count ; ++i)
+        {
+            long now = System.nanoTime();
+            long waitUntil = strategy.computeWaitUntilForContention(1, null, null, null, null);
+            long waitLength = Math.max(waitUntil - now, 0);
+            total += waitLength;
+        }
+        Assert.assertTrue(Math.abs(TimeUnit.MILLISECONDS.toNanos(50) - (total / count)) < TimeUnit.MILLISECONDS.toNanos(1L));
+    }
+
+    @Test
+    public void testBackoffTimeElapsed()
+    {
+        ContentionStrategy strategy = parseStrategy("min=0ms,max=10ms,random=uniform").strategy;
+        double total = 0;
+        int count = 1000;
```

Comment:
Mocking sleep and time would allow measuring if it the calculations are precisely correct.

It's fine for now though.